api/src/main/proto/stellarstation/api/v1/stellarstation.proto

```diff
@@ -0,0 +1,131 @@
+  // this method is called.
+  // If the satellite is not found or the client is not authorized for it, the stream will be closed
+  // with a `NOT_FOUND` error.
+  rpc OpenSatelliteStream (stream SatelliteStreamRequest) returns (stream SatelliteStreamResponse);
+}
+
+// Request for the `OpenSatelliteStreaam` method.
+message SatelliteStreamRequest {
+  // The ID of the satellite to open a stream with. The ID of a satellite can
+  // be found on the StellarStation Console page for the satellite.
+  string satellite_id = 1;
+
+  // A request to send on the satellite stream. Currently only command sending to the satellite is
+  // supported. The request does not need to be filled on the first request to
+  // `OpenSatelliteStream`.
+  oneof Request {
+    // A request to send commands to the satellite.
+    SendSatelliteCommandsRequest send_satellite_commands_request = 2;
+  }
+}
+
+// A request to send commands to a satellite.
+message SendSatelliteCommandsRequest {
+  // The ID of the plan on which to send the commands.
+  //
+  // TODO(anuraag): Add reference to plans API after it is defined.
+  string plan_id = 1;
+
+  // The framing to be applied to `command` before transmission. If `command` has already been
+  // framed, this should be set to RAW and no additional processing will take place before
+  // modulation. When applying framing, each `command` will be treated as a single frame. Framing
+  // parameters used will be the ones registered for the satellite in the StellarStation Console.
+  Framing output_framing = 2;
+
+  // The command frames to send to the satellite. All commands will be transmitted in sequence
+  // immediately, during which time telemetry will not be received for the given `plan_id`. After
+  // all commands have been transmitted, telemetry receive will be immediately re-enabled. The
+  // maximum size of each command is 2MB. If a command larger than 2MB is received, the stream will
+  // be closed with a `RESOURCE_EXHAUSTED` error.
+  repeated bytes command = 3;
+}
+
+// A type of framing of a binary payload used in satellite communicaation.
+enum Framing {
+  // No framing. For sent payloads, this is usually pre-framed data while for received ones, it is
+  // raw IQ data.
+  RAW = 0;
+
+  // AX.25 (Amateur X.25) framing - https://en.wikipedia.org/wiki/AX.25
+  AX25 = 1;
+}
+
+// A response from the `OpenSatelliteStream` method.
+message SatelliteStreamResponse {
+  // A response on the strema. Currently the only type of response is telemetry received from the
+  // satellite.
+  oneof Response {
+    // A response from a satellite containing telemetry.
+    ReceiveTelemetryResponse receive_telemetry_response = 1;
+  }
+}
+
+// A response from a satellite containing telemetry.
+message ReceiveTelemetryResponse {
+  // The ID of the plan on which the telemetry was received.
+  string plan_id = 1;
+
+  // The telemetry received.
+  repeated Telemetry telemetry = 2;
```

Comment:
Does this need to be repeated? I guess telemetry can still be split across SatelliteStreamResponse messages anyway.

Reply:
Yeah I was debating on this - I was thinking of maintaining consistency with `SendCommandRequest`, which does need the ability to send multiple frames, but here I guess singular makes more sense.